### PR TITLE
change how/if TaskName is extracted

### PR DESCRIPTION
**Pull request by julia-pfarr**
so far taskname was extracted from a line in the asc file called *RECORDED_BY* but not all have this line as it is dependent on user settings prior to data collection. If line wasn't there, TaskName appeared in the json with an empty string, which is not useful. Now, if line is not present in asc, TaskName does not appear in json unless the user defined it in the metadata.yml file.

### CODE REVIEW

**Comment by julia-pfarr** [comment]
Actually I think we should just change it to passing it in the metadata.yml file. The ones that have the "RECORDED_BY" entry in the edf usually have the name of the code file they used, like "modif_VWM_task_v2.5_LAB.py" which is probably not the name they would choose for being written in the json. What do you think?

File: eye2bids/_base.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class BasePhysioEventsJson(dict[str, Any]):
 
     input_file: Path
     two_eyes: bool
+    TaskName: bool
```
Comment (Remi-Gau) [inline eye2bids/_base.py:19]: Not sure we need to have this as an attirbute

File: eye2bids/edf2bids.py [hunk 2]
```diff
@@ -520,8 +525,11 @@ def edf2bids(
 
     events_json.input_file = input_file
     events_json.two_eyes = _2eyesmode(df_ms_reduced)
+    events_json.TaskName = _has_TaskName(events)
+
+    if events_json.TaskName:
```
Comment (Remi-Gau) [inline eye2bids/edf2bids.py:530]: I think we can use a wallruss operator here.

Will explain when amin front of a proper computer.

**Review by Remi-Gau — CHANGES_REQUESTED** [review]

File: eye2bids/edf2bids.py [hunk 3]
```diff
@@ -520,8 +525,11 @@ def edf2bids(
...
+        events_json["TaskName"] = _extract_TaskName(events)
```
Comment (Remi-Gau) [inline eye2bids/edf2bids.py:531]:
```suggestion
    if TaskName := _extract_TaskName(events)
        events_json["TaskName"] = TaskName
```

File: eye2bids/edf2bids.py [hunk 4]
```diff
@@ -520,8 +525,11 @@ def edf2bids(
 
     events_json.input_file = input_file
     events_json.two_eyes = _2eyesmode(df_ms_reduced)
+    events_json.TaskName = _has_TaskName(events)
```
Comment (Remi-Gau) [inline eye2bids/edf2bids.py:528]:
```suggestion
```

File: eye2bids/edf2bids.py [hunk 5]
```diff
@@ -254,11 +254,16 @@ def _extract_ScreenResolution(df: pd.DataFrame) -> list[int]:
...
+    TaskName = (
         " ".join([ts for ts in events if ts.startswith("** RECORDED BY")])
         .replace("** RECORDED BY ", "")
         .replace("\n", "")
     )
+    return TaskName
+
+
+def _has_TaskName(events: list[str]) -> bool:
+    return not _extract_TaskName(events) == ""
```
Comment (Remi-Gau) [inline eye2bids/edf2bids.py:266]:
```suggestion
```

File: eye2bids/edf2bids.py (hunk 3 above)
Comment (Remi-Gau) [reply eye2bids/edf2bids.py:531]: walrus operator `:=` in this case is equivalent to 

```python
TaskName = _extract_TaskName(events)
if TaskName:
   ...
```

and the empty string `""` will evaluate as a falsy.

```python
>>> if not  "":
...     print('foo')
... 
foo
```

File: eye2bids/_base.py (hunk 1 above)
Comment (Remi-Gau) [inline eye2bids/_base.py:19]:
```suggestion
```

File: eye2bids/_base.py [hunk 6]
```diff
@@ -48,6 +49,7 @@ def update_from_metadata(self, metadata: None | dict[str, Any] = None) -> None:
             "ScreenRefreshRate": metadata.get("ScreenRefreshRate"),
             "ScreenSize": metadata.get("ScreenSize"),
         }
+        self["TaskName"] = metadata.get("TaskName")
```
Comment (Remi-Gau) [inline eye2bids/_base.py:52]: OK what if the data is both in the metadata and the data file we are reading from: which one takes precedence?
I think we should decide and make that a rule for all similar cases.

**Comment by julia-pfarr** [comment]
@Remi-Gau as discussed earlier I included TaskName as a variable that can exclusively be given through the metadata.yml file and will not be produced by our code.

The changes only take care of the TaskName, I will deal will issue #90 in a separate PR.

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**